odc_k8s/config/flux.yaml

```diff
@@ -9,6 +9,12 @@ git:
   pollInterval: 1m
 registry:
   pollInterval: 1m
+  %{ if flux_registry_ecr.includeId != "" }
+  ecr:
```

Comment:
Also the addition is conditional on having an includeID and I'm not sure that is consistent with the helm chart defaults.

Reply:
here is a helm configuration docs that I referenced - https://github.com/fluxcd/flux/tree/master/chart/flux. The singleton reference work for us as we only have ECR setup in one account and we gave cross account ECR repository access permission. 

Also, not everyone is using ECR so I need to make this config optional.

As per FluxCD/flux reference guild, these are `registry.ecr` parameters -

```
registry.ecr.region | None | Restrict ECR scanning to these AWS regions; if empty, only the cluster's region will be scanned
registry.ecr.includeId | None | Restrict ECR scanning to these AWS account IDs; if empty, all account IDs that aren't excluded may be scanned
registry.ecr.excludeId | 602401143452 | Do not scan ECR for images in these AWS account IDs; the default is to exclude the EKS system account
registry.ecr.require | false | Refuse to start if the AWS API is not available
```

Reply:
Yes, the chart is expects an array which is why its a region**s**, ID**s**, etc.
Also note the default is this can be enabled and it will consider all account IDs and regions that are referenced by ECR urls in the configurations to be valid for scanning.

This TF module though is not consistent with this - only 1 region, ID, etc. Different defaults.
I'm suggesting it should be consistent.

I tested the optional aspect and it works as expected, and also the non-optional one, which requires the additional configuration you pointed me towards.

Reply:
I made necessary changes. please review again